modules/examples/todolist-http-http4s/src/main/scala/todo/http/codecs.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2018 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package examples.todolist
+package http
+
+import cats.effect.Effect
+import io.circe.generic.auto._
+import org.http4s._
+import org.http4s.circe._
+
+object codecs {
+
+  implicit def tagEncoder[F[_]: Effect]: EntityEncoder[F, Tag] = jsonEncoderOf[F, Tag]
```

Comment:
Is the `Effect` instance needed for `jsonEncoderOf`?

Reply:
I have try to avoid it, but I couldn't make it compile. The compilation error I'm getting is:

```scala
[error] /home/jmartinez/workspaces/47deg/freestyle/modules/examples/todolist-http-http4s/src/main/scala/todo/http/codecs.scala:27:79: Cannot convert from String to an Entity, because no EntityEncoder[F, String] instance could be found.
[error]   implicit def tagEncoder[F[_]]: EntityEncoder[F, Tag]         = jsonEncoderOf[F, Tag]
```

Reply:
I found it. [`jsonEncoder` requires an `Applicative`](https://github.com/http4s/http4s/blob/f5538d84067408ad7a4ce78b4d92985df537928a/circe/src/main/scala/org/http4s/circe/CirceInstances.scala#L94) and [`jsonOf` a `Sync` instance](https://github.com/http4s/http4s/blob/f5538d84067408ad7a4ce78b4d92985df537928a/circe/src/main/scala/org/http4s/circe/CirceInstances.scala#L48). So, please, could you change it for expressing this constraint? Thanks

```scala
  implicit def tagEncoder[F[_]: Applicative]: EntityEncoder[F, Tag] = jsonEncoderOf[F, Tag]
  implicit def tagDecoder[F[_]: Sync]: EntityDecoder[F, Tag]        = jsonOf[F, Tag]

  implicit def todoItemEncoder[F[_]: Applicative]: EntityEncoder[F, TodoItem] =
    jsonEncoderOf[F, TodoItem]
  implicit def todoItemDecoder[F[_]: Sync]: EntityDecoder[F, TodoItem] = jsonOf[F, TodoItem]

  implicit def todoListEncoder[F[_]: Applicative]: EntityEncoder[F, TodoList] =
    jsonEncoderOf[F, TodoList]
  implicit def todoListDecoder[F[_]: Sync]: EntityDecoder[F, TodoList] = jsonOf[F, TodoList]

  implicit def todoFormEncoder[F[_]: Applicative]: EntityEncoder[F, TodoForm] =
    jsonEncoderOf[F, TodoForm]
  implicit def todoFormDecoder[F[_]: Sync]: EntityDecoder[F, TodoForm] =
    jsonOf[F, TodoForm]
```